src/health.rs

```diff
@@ -0,0 +1,74 @@
+use std::collections::HashMap;
+
+use crate::Actuator;
+use rocket::{http::Status, serde::json::Json};
+use serde::Serialize;
+
+pub mod check;
+
+#[derive(Serialize)]
+pub enum IndicatorStatus {
+    Up,
+    Down,
+}
+
+#[derive(Serialize)]
+pub struct IndicatorResult {
+    status: IndicatorStatus,
+    message: Option<String>,
+}
+
+impl From<Result<(), String>> for IndicatorResult {
+    fn from(value: Result<(), String>) -> Self {
+        match value {
+            Ok(()) => Self {
+                status: IndicatorStatus::Up,
+                message: None,
+            },
+            Err(m) => Self {
+                status: IndicatorStatus::Down,
+                message: Some(m),
+            },
+        }
+    }
+}
+
+#[async_trait::async_trait]
+pub trait Indicator: Send + Sync {
+    fn name(&self) -> &str;
+    async fn check(&self) -> Result<(), String>;
+}
+
+#[async_trait::async_trait]
+pub trait Check {
+    fn indicators(&self) -> &Vec<Box<dyn Indicator>>;
+
+    async fn check(
+        &self,
+    ) -> rocket::response::status::Custom<Json<HashMap<&str, IndicatorResult>>> {
+        let mut status = Status::Ok;
+
+        let mut results: HashMap<&str, IndicatorResult> =
+            HashMap::with_capacity(self.indicators().len());
+
+        for i in self.indicators() {
```

Comment:
Bu indicator'ları paralel çalıştırmak daha iyi olmaz mı?

Reply:
Buraya pek timeout ekleme taraftarı değilim. Timeout eklenecek ise indicator'ların içine eklenmesi gerektiğini düşünüyorum. Her indicator'un kendine özgü timeout'u olabilir.

Reply:
Checks are now performed in parallel.